cluster/ubuntu/util.sh

```diff
@@ -369,6 +374,15 @@ function provision-master() {
     ubuntu/binaries/master/ \
     "${MASTER}:~/kube"
 
+  if [ -z "$CNI_PLUGIN_CONF" ] || [ -z "$CNI_PLUGIN_EXES" ]; then
+    # Flannel is being used: copy the flannel binaries and scripts, set reconf flag
+    scp -r $SSH_OPTS ubuntu/master-flannel/* "${MASTER}:~/kube"
+    NEED_RECONFIG_DOCKER=true
+  else
+    # CNI is being used: set reconf flag
```

Comment:
>Regarding tarball URL instead of file pathname set: the added generality is good. Do you know a way to secure the tar xf command so that it doesn't write in unwanted places?

Unsure, but just encode the tarball url into the code as default url and make sure that that tarball doesn't write to random locations sounds Ok to me. I just want to give people an easy way to get off the ground with their CNI plugin and local-up/ubuntu setup. Plugin installation is slightly convoluted in kube for legacy reasons, and hence error prone. 

> As for sanity checking, it seems a little inconsistent to check that the CNI_PLUGIN_EXES exist but not also check that the CNI_PLUGIN_CONF exists.

If you're interested in, eg: parsing out the type of the plugin from the cni.conf and checking that there is an exec with the same name in the bin dir, you could use jq (eg: https://github.com/appc/cni/blob/master/scripts/exec-plugins.sh#L16), but you don't necessarily need to do so in this pr. 

> BTW, if we switch to one tarball URL for the executables then we could also get the CNI config file from the tarball, reducing two config envars to one.

SGTM

> BTW, switching from file pathname set to one tarball does not eliminate the need for checking; it imposes a need for certain structure in the tarball contents and checking for conformance to that.

Yeah I think you can assume your default tarballs follows this structure, and assume that they're on their own if they specify another url. 

> Do we really need this much checking? If so, can we do it as a later PR that builds on this one?

Yeah, I suppose checking could get hairy. I dont' really want "checking" as much as "you're asking for CNI but your plugins and conf are in the wrong place" type failure message.